roachtest:  de-flake multitenant-fairness/read-heavy/skewed

irfansharif opened the pull request:
Fixes #97448 (possibly).
Fixes #78691.

These tests run under severe CPU overload, and we see the workload
getting observing the following errors:

  ERROR: liveness session expired 571.043163ms before transaction

The SQL liveness lease extension work ends up getting severely starved,
despite extending leases by 40s every 5s. It turns out for tenant
SQL liveness work, we were using admissionpb.NormalPri, so such
starvation was possible. This wasn't true for the system tenant where we
bypassed AC altogether.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98785)
<!-- Reviewable:end -->

irfansharif committed:
roachtest: de-flake multitenant-fairness/read-heavy/skewed

Fixes #97448 (possibly).
Fixes #78691.

These tests run under severe CPU overload, and we see the workload
getting observing the following errors:

  ERROR: liveness session expired 571.043163ms before transaction

The SQL liveness lease extension work ends up getting severely starved,
despite extending leases by 40s every 5s. It turns out for tenant
SQL liveness work, we were using admissionpb.NormalPri, so such
starvation was possible. This wasn't true for the system tenant where we
bypassed AC altogether.

Release note: None

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98785#-:-NR3fY-G2zb_WPLz1JVm:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @fqazi, @irfansharif, and @sumeerbhola)

___
*[`pkg/sql/sqlliveness/slstorage/slstorage.go` line 312 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98785#-NR3f9Og-ovd68UsErd2:-NR3f9Og-ovd68UsErd3:b-6vnyjz) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e855165febf62bd553cb3627ae5919dcd6c958b/pkg/sql/sqlliveness/slstorage/slstorage.go#L312)):*
> ```Go
> // appropriate AC metadata for SQL liveness work.
> func (s *Storage) txn(ctx context.Context, retryable func(context.Context, *kv.Txn) error) error {
> 	// We use AdmissionHeader_OTHER to bypass AC if we're the system tenant
> ```

I see this is overwritten downstream to the correct thing, but at this level it's slightly surprising to read this comment and not see a conditional to check if we're the system tenant. Let's add a few words about why we're using AdmissionHeader_OTHER for secondary tenant's as well? 


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98785)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @irfansharif, and @sumeerbhola)



<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98785#-:-NR3jOJ9EUi7fQuTgNC6:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @ajwerner, @irfansharif, and @sumeerbhola)



<!-- Sent from Reviewable.io -->
